src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -314,7 +314,7 @@ bbf_selectDumpableObject(DumpableObject *dobj, Archive *fout)
 				ExtensionInfo	*ext = findOwningExtension(cast->dobj.catId);
 
 				/* Skip if cast is not a member of babelfish extension */
-				if (ext == NULL || strcmp(ext->dobj.name, "babelfishpg_common") != 0)
+				if (ext == NULL || strncmp(ext->dobj.name, "babelfishpg_common", 18) != 0)
```

Comment:
what is the rationale behind this? with this any name with babelfish_common as prefix will match, is that desired beahvior?

Reply:
The main reason was that `strcmp` is considered unsafe c function and would come in the security findings but I just looked at the tool again and looks like it does not consider this as unsafe.

Reply:
using strncmp is ok but if we do not compare length, we can get incorrect semantics.
strcmp(str1, str2) is not same as strncmp(str1, str2, strlen(str2) (str1 and str2 will match as long as str2 is a substring of str1